components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -951,4 +952,19 @@ private static boolean isResourceEpUnavailable(Operation operation) {
         }
         return false;
     }
+
+    public static List<String> setAuthProviders(BasicAuth basicAuth) {
+        List<String> authProviders = new ArrayList<>();
+        if (basicAuth != null && basicAuth.isOptional()) {
+            authProviders.add(APISecurity.basic.name());
+            authProviders.add(APISecurity.oauth2.name());
+            authProviders.add(APISecurity.jwt.name());
+        } else if (basicAuth != null && basicAuth.isRequired()) {
+            authProviders.add(APISecurity.basic.name());
+        } else {
```

Comment:
Lets remove this logic. We need to honour the values comes from the swagger.  If the security definitions are not defined by default oauth and jwt should be added.

Reply:
Fixed in #932